pkg/kubectl/cmd/run.go

```diff
@@ -213,7 +214,9 @@ func Run(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cobr
 		}
 		switch restartPolicy {
 		case api.RestartPolicyAlways:
-			if contains(resourcesList, v1beta1.SchemeGroupVersion.WithResource("deployments")) {
+			if contains(resourcesList, appsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
+				generatorName = "deployment/apps.v1beta1"
+			} else if contains(resourcesList, extensionsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
 				generatorName = "deployment/v1beta1"
```

Comment:
Since this already gets a release note we can mention the behavior change but I don't think it will actually break anybody.